src/components/modules/dragNDrop.ts

```diff
@@ -75,16 +65,43 @@ export default class DragNDrop extends Module {
 
     dropEvent.preventDefault();
 
+    if (BlockManager.currentDraggingBlock) {
+      const currentIndex = this.Editor.BlockManager.currentDraggingBlockIndex;
+      const targetBlock = BlockManager.getBlockByChildNode(dropEvent.target as Node);
+
+      if(!targetBlock) {
+        //This means that we are trying to drop a block without references.
+        return;
+      }
+
+      const targetIndex = this.Editor.BlockManager.blocks.findIndex(block => block === targetBlock);
+      if (targetBlock.dropTargetPlacement === 'top') {
+        if (targetIndex > currentIndex) {
+          this.Editor.BlockManager.move(targetIndex - 1);
+        } else {
+          this.Editor.BlockManager.move(targetIndex);
+        }
+      } else if (targetBlock.dropTargetPlacement === 'bottom') {
+        if (targetIndex > currentIndex) {
+          this.Editor.BlockManager.move(targetIndex);
+        } else {
+          this.Editor.BlockManager.move(targetIndex + 1);
+        }
+      }
+
+      //this has to be cleaned after we drop the block
+      BlockManager.blocks.forEach((block) => {
+        block.dropTarget = false;
+      });
+
+      return;
+    }
+
+    //this has to be cleaned after the try/catch
     BlockManager.blocks.forEach((block) => {
       block.dropTarget = false;
     });
 
-    if (SelectionUtils.isAtEditor && !SelectionUtils.isCollapsed && this.isStartedAtEditor) {
```

Comment:
why this statement has been removed?

Reply:
Sorry, I think this conflicted with my first approach and I forgot to restore it.